deflate_p.h

```diff
@@ -33,7 +33,10 @@ static inline int zng_tr_tally_lit(deflate_state *s, unsigned char c) {
     s->sym_buf[s->sym_next++] = c;
     s->dyn_ltree[c].Freq++;
     Tracevv((stderr, "%c", c));
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtype-limits"
```

Comment:
Why these three lines for GCC only? Is there another way this can be done through casting?

Reply:
Casts don't work in macro parameters as in most cases macro parameters need to be a single token. This is also why in some macros the macro parameters are assigned to variables before use, or there is extra parentheses when accessing macro parameters. In preprocessor, token starts with "(" and ends with ")" or is completely alphanumeric.... 

Some documents say preprocessor tokens can contain also "_" but the result is undefined if there is ambiguity, as you can pretty much rename or split any preprocessor token by defining another token whose name is substring of the name of the first token.

When preprocessor parses a token that results in a string literal, space (ASCII 32) is appended to the output unless the token is followed by "##", which is concatenation operator in preprocessor.